src/App.js

```diff
@@ -0,0 +1,198 @@
+  postCardAsync,
+  likeCardAsync,
+  deleteCardAsync,
+  selectBoardAsync,
+  getAllBoardsAsync,
+} from "./apiCalls";
+
+function App() {
+  const [boardData, setBoardData] = useState([]);
+  const [selectedBoardId, setSelectedBoardId] = useState(null);
+  const [errorMessage, setErrorMessage] = useState(null);
+
+  const displayAllBoards = () => {
+    getAllBoardsAsync()
+      .then((boards) => {
+        setBoardData(boards);
+      })
+      .catch((err) => {
+        console.log(err);
+        setErrorMessage("error displaying all boards");
+      });
+  };
+
+  useEffect(() => {
+    displayAllBoards();
+  }, [boardData]);
+
+  const selectBoard = (id) => {
+    selectBoardAsync(id)
+      .then((board) => {
+        setSelectedBoardId(board.boardId);
+      })
+      .catch((err) => {
+        console.log(err.message);
+        setErrorMessage("error displaying board");
+      });
+  };
+
+  const postBoard = (boardData) => {
+    postBoardAsync(boardData)
+      .then((newBoard) => {
+        setBoardData((oldData) => [...oldData, newBoard]);
+      })
+      .catch((err) => {
+        console.log(err.message);
+        setErrorMessage(err.message);
+      });
+  };
+
+  const postCard = (cardData) => {
+    postCardAsync(cardData, selectedBoardId)
+      .then(() => {
+        setBoardData((oldBoardData) => {
+          const newBoardData = oldBoardData.map((board) => {
+            if (board.boardId === selectedBoardId) {
+              return getSelectedBoard(selectedBoardId);
+            } else {
+              return board;
+            }
+          });
+          return newBoardData;
+        });
+      })
+      .catch((err) => {
+        console.log(err.message);
+        setErrorMessage(err.message);
+      });
+  };
+
+  const likeCard = (cardId) => {
+    likeCardAsync(cardId, selectedBoardId)
+      .then(() => {
+        setBoardData((oldBoardData) => {
+          const newBoardData = oldBoardData.map((board) => {
+            if (board.boardId === selectedBoardId) {
+              return getSelectedBoard(selectedBoardId);
+            } else {
+              return board;
+            }
+          });
+          return newBoardData;
+        });
+      })
+      .catch((err) => {
+        console.log(err.message);
+        setErrorMessage(err.message);
+      });
+  };
+
+  const deleteCard = (cardId) => {
+    deleteCardAsync(cardId, selectedBoardId)
+      .then(() => {
+        setBoardData((oldBoardData) => {
+          const newBoardData = oldBoardData.map((board) => {
+            if (board.boardId === selectedBoardId) {
+              return getSelectedBoard(selectedBoardId);
+            } else {
+              return board;
+            }
+          });
+          return newBoardData;
+        });
+      })
+      .catch((err) => {
+        console.log(err.message);
+        setErrorMessage(err.message);
+      });
+  };
+
+  const handleNewBoard = (formFields) => {
+    postBoard(formFields);
+  };
```

Comment:
I'd consider if `handleNewBoard` is necessary and, since they take the same arguments, if we could pass the function `postBoard` directly anywhere `handleNewBoard` was being used.